File: Python/Prodi.py
from Course import Course

class Prodi:
    __nama_prodi = ""
    __kode = ""
    __fakultas = ""
    __courses = []
    
    def __init__(self, nama_prodi, kode, fakultas, course):
        self.__nama_prodi = nama_prodi
        self.__kode = kode
        self.__fakultas = fakultas
        self.__courses.append(course)
    
    def set_nama_prodi(self, nama_prodi):
        self.__nama_prodi = nama_prodi
    
    def set_dosen(self, kode):
        self.__kode = kode
        
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
    
    def add_courses(self, course):
        self.__courses.append(course)
    
    def get_nama_prodi(self):
        return self.__nama_prodi
    
    def get_kode(self):
        return self.__kode
    
    def get_fakultas(self):
        return self.__fakultas
    
    def get_courses(self):
        return self.__courses

File: C++/Prodi.cpp
/* Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543 mengerjakan evaluasi Latihan 4
dalam mata kuliah Design Pemrograman Berorientasi Objek 2023
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

#include "header.h"

// Fungsi construct kosong
Prodi::Prodi()
{
    this->nama_prodi = "";
    this->kode = "";
    this->fakultas = "";
    this->courses.clear();
}

// Fungsi construct dengan parameter
Prodi::Prodi(string nama_prodi, string kode, string fakultas, Course course)
{
    this->nama_prodi = nama_prodi;
    this->kode = kode;
    this->fakultas = fakultas;
    this->courses.clear();
    this->courses.push_back(course);
}

// Setter dan getter tiap atribut yang ada di class ini
void Prodi::set_nama_prodi(string nama_prodi)
{
    this->nama_prodi = nama_prodi;
}

void Prodi::set_kode(string kode)
{
    this->kode = kode;
}

void Prodi::set_fakultas(string fakultas)
{
    this->fakultas = fakultas;
}

void Prodi::add_courses(Course course)
{
    this->courses.push_back(course);
}

string Prodi::get_nama_prodi()
{
    return this->nama_prodi;
}

string Prodi::get_kode()
{
    return this->kode;
}

string Prodi::get_fakultas()
{
    return this->fakultas;
}

list<Course> Prodi::get_courses()
{
    return this->courses;
}

Prodi::~Prodi()
{
}